httomo/task_runner.py

```diff
@@ -216,30 +263,22 @@ def run_tasks(
         start = time.perf_counter_ns()
 
         # check if the module needs the ncore parameter and add it
-        if "ncore" in signature(func_method).parameters:
-            dict_params_method.update({"ncore": ncore})
+        if "ncore" in signature(method_func.method_func).parameters:
+            method_func.parameters.update({"ncore": ncore})
 
         idx += 1
-        reslice_counter, has_reslice_warn_printed, glob_stats[idx] = _run_method(
+        reslice_info, glob_stats[idx] = _run_method(
             idx,
             save_all,
-            module_path,
-            package,
-            method_name,
-            dict_params_method,
             possible_extra_params,
-            len(method_funcs),
-            func_wrapper,
-            method_funcs[idx][1],
-            method_funcs[idx - 1][1],
+            method_func,
+            method_funcs[idx - 1],
+            method_funcs[idx + 1] if idx < len(method_funcs) - 1 else None,
             dict_datasets_pipeline,
             httomo.globals.run_out_dir,
```

Comment:
no need to have this as an argument to the function.

Reply:
thought: see if having a data class `PipelineInfo` would be useful that contains some general info about the pipeline, like `save_all`, `no_of_tasks` etc - info that is shared by all/some of the methods.

Reply:
todo: add tests for `MethodFunc`.